app/validation/validator.py

```diff
@@ -302,25 +302,29 @@ def validate_when_rule(when_clause, answer_ids_with_group_id, referenced_id):
         Validates any answer id in a when clause exists within the schema
         Will also check that comparison_id exists
         """
-        answer_reference_id_fields = ('id', 'answer_count', 'comparison_id')
+        answer_reference_id_fields = ('id', 'answer_ids', 'answer_count', 'comparison_id')
 
         for when in when_clause:
             present_id_ref_keys = [x for x in answer_reference_id_fields if x in when]
-
             for id_ref_key in present_id_ref_keys:
+                # answer_ids is a list, so we have to handle it a little differently to get the same error message.
+                if id_ref_key == 'answer_ids':
+                    for answer_id in when['answer_ids']:
+                        if answer_id not in answer_ids_with_group_id:
+                            return Validator._error_message('The answer id - {} in the {} key of the "when" clause for {} does not exist'
+                                                            .format(answer_id, id_ref_key, referenced_id))
+                else:
+                    if when[id_ref_key] not in answer_ids_with_group_id:
+                        return Validator._error_message('The answer id - {} in the {} key of the "when" clause for {} does not exist'
+                                                        .format(when[id_ref_key], id_ref_key, referenced_id))
 
-                if when[id_ref_key] not in answer_ids_with_group_id:
-                    return Validator._error_message('The answer id - {} in the {} key of the "when" clause for {} does not exist'
-                                                    .format(when[id_ref_key], id_ref_key, referenced_id))
-
-                if id_ref_key == 'answer_count':
+                if id_ref_key == 'answer_count' or when.get('type') == 'answer_count':
```

Comment:
This should be moved out of the loop - it doesn't make sense in the loop of `present_id_ref_keys` as the when type does not contain an answer id.